pydepsi/io.py

```diff
@@ -0,0 +1,351 @@
+
+    # ++++ 29 - n_azimuth
+    if crop_flag:
+        crop_file = "/".join(str(resfile).split("/")[0:-2]) + "/nlines_crp.txt"
+        with open(crop_file) as file:
+            content = file.readlines()
+            n_lines, first_line, last_line = (
+                int(content[0].strip()),
+                int(content[1].strip()),
+                int(content[2].strip()),
+            )
+
+    else:
+        # Extract first
+        pattern = r"First_line \(w.r.t. original_image\):" + SC_N_PATTERN
+        match = re.search(pattern, content)
+        first_line = int(match.group(1))
+        # Extract last
+        pattern = r"Last_line \(w.r.t. original_image\):" + SC_N_PATTERN
+        match = re.search(pattern, content)
+        last_line = int(match.group(1))
+        # difference
+        n_lines = last_line - first_line + 1
+
+    # ++++ 30 - n_range
+    if crop_flag:
+        crop_file = "/".join(str(resfile).split("/")[0:-2]) + "/npixels_crp.txt"
+        with open(crop_file) as file:
+            content = file.readlines()
+            n_pixels, first_pixel, last_pixel = (
+                int(content[0].strip()),
+                int(content[1].strip()),
+                int(content[2].strip()),
+            )
+    else:
+        # Extract first
+        pattern = r"First_pixel \(w.r.t. original_image\):" + SC_N_PATTERN
+        match = re.search(pattern, content)
+        first_pixel = int(match.group(1))
+        # Extract last
+        pattern = r"Last_pixel \(w.r.t. original_image\):" + SC_N_PATTERN
+        match = re.search(pattern, content)
+        last_pixel = int(match.group(1))
+        # difference
+        n_pixels = last_pixel - first_pixel + 1
+
+    # ----------------------------------------
+
+    # Fill the dictionary
+    datewise_metadata = {
+        "orbit": geometry,
+        "acq_date": acq_date,
+        "azimuth0time": azimuth0time,
+        "range0time": range0time / 2,
+        "prf": prf,
+        "rsr": rsr,
+        "wavelength": wavelength,
+        "orbit_fit": orbfit,
+        "range_spacing": range_spacing,
+        "azimuth_spacing": azimuth_spacing,
+        "center_lon": center_lon,
+        "center_lat": center_lat,
+        "center_h": center_h,
+        "n_azimuth": n_azimuth,
+        "n_range": n_range,
+        "1stAzimuth": first_line,
+        "1stRange": first_pixel,
+        "swath": swath,
+        "center_azimuth": center_azimuth,
+        "beta0": beta0,
+        "azimuth_resolution": azimuth_resolution,
+        "range_resolution": range_resolution,
+        "nBursts": 1,
+        "burstInfo": burst_n,
+        "steering_rate": steering_rate,
+        "pri": pri,
+        "rank": rank,
+        "chirp_rate": chirp_rate,
+        "n_lines": n_lines,
+        "n_pixels": n_pixels,
+        # -------------------------------------------------------------------
+    }
+
+    return datewise_metadata
+
+
+def _orbit_fit(orbit, verbose=0, plot_flag=0, der=True):
```

Comment:
orbit fitting does't seem directly i/o related, although I see it is used to derive parameters which I assume are then used later, so akin to reading in. I'd suggest to spin the function out, and have it called here to populate the orbit_fit if that is used. Otherwise maybe not in io.py

Reply:
I agree. For now, I separate this function to `utils.py`